packages/yoroi-extension/app/components/wallet/create-wallet/VerifyRecoveryPhraseStep.js

```diff
@@ -0,0 +1,239 @@
+// @flow
+import { useState, useMemo } from 'react';
+import type { Node, ComponentType } from 'react';
+import { defineMessages, injectIntl, FormattedHTMLMessage } from 'react-intl';
+import { observer } from 'mobx-react';
+import type { $npm$ReactIntl$IntlShape } from 'react-intl';
+import { Stack, Box, Typography, Button } from '@mui/material';
+import StepController from './StepController';
+import { CREATE_WALLET_SETPS } from './steps';
+import styles from './VerifyRecoveryPhraseStep.scss';
+import classnames from 'classnames';
+import { ReactComponent as VerifiedIcon } from '../../../assets/images/verify-icon-green.inline.svg';
+import environment from '../../../environment';
+import { makeSortedPhrase } from '../../../utils/recoveryPhrase';
+import globalMessages from '../../../i18n/global-messages';
+
+const messages = defineMessages({
+  description: {
+    id: 'wallet.create.thirdStep.description',
+    defaultMessage:
+      '!!!<strong>Select</strong> each word in <strong>the correct order</strong> to confirm your recovery phrase.',
+  },
+  incorrectOrder: {
+    id: 'wallet.create.thirdStep.incorrectOrder',
+    defaultMessage: '!!!Incorrect order. Try again',
+  },
+  verified: {
+    id: 'walllet.create.thirdStep.verifiedRecoveryPhrase',
+    defaultMessage: '!!!The recovery phrase is verified',
+  },
+});
+
+type Intl = {|
+  intl: $npm$ReactIntl$IntlShape,
+|};
+
+type Props = {|
+  setCurrentStep(stepId: string): void,
+  recoveryPhrase: Array<string> | null,
+|};
+
+function VerifyRecoveryPhraseStep(props: Props & Intl): Node {
+  const { intl, recoveryPhrase, setCurrentStep } = props;
+  if (!recoveryPhrase) throw new Error('Missing recovery phrase, should never happen');
+
+  const [enteredRecoveryPhrase, setRecoveryPhrase] = useState(
+    new Array(recoveryPhrase.length).fill(null)
+  );
+  const [wrongWord, setWrongWord] = useState<string | null>(null);
+
+  function onAddWord(word: string, idx: number): void {
+    if (isWordAdded(word)) return;
+
+    const nextWordIdx = enteredRecoveryPhrase.findIndex(w => w === null);
+    if (nextWordIdx === -1) throw new Error('Entered recovery phrase words list is full');
+
+    const isInCorrectOrder = recoveryPhrase[nextWordIdx] === word;
+    if (!isInCorrectOrder) {
+      return setWrongWord(word);
+    }
+
+    setRecoveryPhrase(prev => {
+      const copy = [...prev];
+      copy[idx] = word;
+      return copy;
+    });
+    setWrongWord(null);
+  }
+
+  function isWordAdded(word) {
+    return enteredRecoveryPhrase.some(w => w === word);
+  }
```

Comment:
@AhmedIbrahim336 , recovery phrases can and will contain multiple instances of the same word , you cannot decide only by the word itself that a similar word should not be entered again. For example, this is a valid bip39 mnemonic:
```
day day day day day day day day day day day day day day day
```

Always keep this case in mind.